frontend/src/components/TextEditor.jsx

```diff
@@ -24,12 +24,30 @@ export default function TextEditor({ document = [], onChange}) {
     initializeStateWithAllCommentThreads(editor, addCommentThread);
   }, [editor, addCommentThread]);
 
+  const onChange = (value) => {
+    updateDocumentState(value)
+    const isAstChange = editor.operations.some(
+      op => 'set_selection' !== op.type
+    )
+    if (isAstChange) {
+      // Save the value to Local Storage.
+      const content = JSON.stringify(value)
+      localStorage.setItem('content', content)
+    }
+  } 
+
+  const initialValue = useMemo(
```

Comment:
I don't think you need to change this, the initial value of the document is already read in from localstore in the `Document` component and passed into here. You just use it to initialize the editor, and then the editor state will always match the localstore state after initialization anyways? So I feel like you should just be able to remove this `initialValue` and go back to passing in `document` in line 47?